app/renderer/reducers/contextMenuReducer.js

```diff
@@ -89,6 +101,414 @@ const onTabPageMenu = function (state, action) {
   tabPageMenu.popup(getCurrentWindow())
 }
 
+const openInNewTabMenuItem = (url, isPrivate, partitionNumber, openerTabId) => {
```

Comment:
Each of these menu types should be in a file (I'd suggest commonMenu- see https://github.com/brave/browser-laptop/blob/2bd7726b6d6ea83f41fbebdfab6b4141f8a5a78b/app/common/commonMenu.js). This would make it easy to unit test each method.  A quick example of something easy to test: you can check each method that's exported to make sure each has a label field that called `locale.translation()`.